cg/meta/compress/compress.py

```diff
@@ -259,27 +264,38 @@ def update_fastq_hk(
             hk_fastq_first=hk_fastq_first, hk_fastq_second=hk_fastq_second
         )
 
-    def add_fastq_hk(self, sample_obj: Sample, fastq_first: Path, fastq_second: Path) -> None:
-        """Add FASTQ files to Housekeeper."""
-
-        if not sample_obj.application_version.application.is_external:
-            flow_cell_id: str = self.get_flow_cell_id(fastq_path=fastq_first)
-            fastq_tags: List[str] = [flow_cell_id, SequencingFileTag.FASTQ]
-        else:
-            fastq_tags: List[str] = [sample_obj.internal_id, SequencingFileTag.FASTQ]
+    def add_fastq_hk(
+        self,
+        spring_path: Path,
+        sample_internal_id: str,
+        fastq_first: Path,
+        fastq_second: Path,
+    ) -> None:
+        """Add decompressed FASTQ files to Housekeeper."""
+        spring_file: File = self.hk_api.files(path=spring_path.as_posix()).first()
+        fastq_tags = [
+            SequencingFileTag.FASTQ if tag_name == SequencingFileTag.SPRING else tag_name
```

Comment:
I like this, but in general, we try to avoid nested loops in list comprehensions. I suggest creating the list of spring tags, removing the spring and replacing it with fastq as you did in the previous PR

Reply:
Split it into a separate method and went with your approach.